### PR TITLE
[FP-29] Custom List Component

### PR DESCRIPTION
**Pull request by thisisofficial**
### Description

- Added the list component. It take an array of any kind and returns a formatted list.
- Added a list item component, It takes a string or number as a title and an extra, returning row for the list component.

Both are expandable and can take infinite ammounts of data, as long as it is formatted as an array.

image

- It can also be somewhat styled using a prop that changes the width.
image

### Testing Checklist

- [x] Test cases have been added/updated
- [x] Tested locally in browser (at least one other than Chrome)
- [x] All tests pass

### What JIRA Ticket(s) does this PR address?

- [FP-29](https://cd95.atlassian.net/browse/FP-29)

### CODE REVIEW

**Comment by thisisofficial** [comment]
I want to get comments on my current progress, I'm currently missing one file that will be completed shortly...regardless, please review this as if that file wasn't there.

File: src/components/listItems/listItem.scss [hunk 1]
```diff
@@ -0,0 +1,37 @@
+.listitem{
```
Comment (caarlosdamian) [inline src/components/listItems/listItem.scss]: use kebab-case
in all the classnames

File: src/components/listItems/listItem.scss [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    display: flex;
+    flex-direction: row;
```
Comment (caarlosdamian) [inline src/components/listItems/listItem.scss]: row is default of display flex , remove this property

File: src/components/list/list.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { ListItem } from '../listItems/listItem';
+
+interface Prop {
+  obj: Array<Array<number | string>>;
+}
+
+export const List = ({ obj }: Prop) => (
+  <div className="list-container">
```
Comment (caarlosdamian) [inline src/components/list/list.tsx]: this class is not styled anywhere

**Review by caarlosdamian — CHANGES_REQUESTED** [review]
Please attached a video or screenshots on how the component behaves

File: src/components/list/list.scss [hunk 4]
```diff
@@ -0,0 +1,9 @@
+.list-container{
+    border: 1px gray solid;
```
Comment (caarlosdamian) [inline src/components/list/list.scss]: Border is not the same 
expected 

Screen Shot 2022-09-26 at 9 54 48

actual 
image

File: src/components/listItems/listItem.scss [hunk 5]
```diff
@@ -0,0 +1,36 @@
+.list-item{
```
Comment (caarlosdamian) [inline src/components/listItems/listItem.scss]: Desing is not matching figma desing 

- Fonts are not equal 
- Line between items not require 
- Size of spacing not the same 
-  color of the fonts 
- not suppose to be a Card is only the list a card component is already created 
- extra border on last item remove it 
- item should be exact as figma desing 
Expected 
Screen Shot 2022-09-26 at 9 56 12
actual 

image

**Review by caarlosdamian — CHANGES_REQUESTED** [review]

File: src/components/list/list.tsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import './list.scss';
+
+interface Prop {
+  obj: Array<Array<number | string>>;
+  width?: string | number | undefined;
+  height?: string | number | undefined;
+}
+
+export const List = ({ obj, width, height }: Prop) => (
+  <div
```
Comment (caarlosdamian) [inline src/components/list/list.tsx]: Test of the component is missing

**Review by caarlosdamian — DISMISSED** [review]

File: src/components/list/list.tsx [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+import { ListItem } from '../listItems/listItem';
+import './list.scss';
+
+interface Prop {
+  obj: Array<Array<number | string>>;
+  width?: string | number | undefined;
+  height?: string | number | undefined;
+}
+
+export const List = ({ obj, width, height }: Prop) => (
```
Comment (caarlosdamian) [inline src/components/list/list.tsx]: Component files always need to be Capitalize and match scss file 
List.tsx
List.scss

Comment (caarlosdamian) [reply src/components/list/list.tsx]: src/components/list/List.tsx for example

File: src/components/list/list.tsx [hunk 8]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { ListItem } from '../listItems/listItem';
+import './list.scss';
+
+interface Prop {
+  obj: Array<Array<number | string>>;
```
Comment (caarlosdamian) [inline src/components/list/list.tsx]: be more verbose obj is not good  to describe what it is


if is and Array I sugguest `items`

File: src/components/list/list.tsx [hunk 9]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { ListItem } from '../listItems/listItem';
+import './list.scss';
+
+interface Prop {
```
Comment (caarlosdamian) [inline src/components/list/list.tsx]: change name to `ListProps`

File: src/components/listItems/listItem.scss [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+    display: flex;
+    min-width: 100px;
+    width: fit-content;
+    width: 100%;
+    font-family: 'Mulish';
+    font-style: normal;
+    font-weight: 600;
+    padding: 3px;
+    .item-title{
+        max-width: 50%;
```
Comment (caarlosdamian) [inline src/components/listItems/listItem.scss]: use flex 1

File: src/components/listItems/listItem.scss [hunk 11]
```diff
@@ -0,0 +1,35 @@
+.list-item{
+    display: flex;
```
Comment (caarlosdamian) [inline src/components/listItems/listItem.scss]:
```
display:flex;
justify-content:space-between; should do all the work 
align-items:center
```

File: src/components/list/list.tsx [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+  obj: Array<Array<number | string>>;
+  width?: string | number | undefined;
+  height?: string | number | undefined;
+}
+
+export const List = ({ obj, width, height }: Prop) => (
+  <div
```
Comment (caarlosdamian) [inline src/components/list/list.tsx]: this should be and 

`ul ` element

File: src/components/listItems/listItem.tsx [hunk 13]
```diff
@@ -0,0 +1,24 @@
+import './listItem.scss';
+
+interface Props {
+  title: string | number;
+  item?: string | number;
+  last?: boolean;
+}
+
+export const ListItem = ({ title, item, last }: Props) => (
+  <div
```
Comment (caarlosdamian) [inline src/components/listItems/listItem.tsx]: this should be a `li` element

File: src/components/listItems/listItem.tsx [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+    className="list-item"
+    style={{ borderBlockEnd: last ? 'none' : '2px solid #DFE0EB' }}
```
Comment (caarlosdamian) [inline src/components/listItems/listItem.tsx]: be more verbose 
`itemTitle` -> should only be a string 
`lastItem` 


what does   `item?: string | number; ` mean ??

File: src/components/listItems/listItem.tsx [hunk 15]
```diff
@@ -0,0 +1,24 @@
+import './listItem.scss';
+
+interface Props {
+  title: string | number;
+  item?: string | number;
+  last?: boolean;
+}
+
+export const ListItem = ({ title, item, last }: Props) => (
```
Comment (caarlosdamian) [inline src/components/listItems/listItem.tsx]: Same here name of file tsx and scss should be Capitalize

File: src/components/listItems/listItem.tsx [hunk 16]
```diff
@@ -0,0 +1,24 @@
+import './listItem.scss';
+
+interface Props {
```
Comment (caarlosdamian) [inline src/components/listItems/listItem.tsx]: ListItemProps

File: src/components/list/list.scss [hunk 17]
```diff
@@ -0,0 +1,5 @@
+.list-container{
+    border-radius: 10px;
```
Comment (caarlosdamian) [inline src/components/list/list.scss]: the container should be in another pr

File: src/components/list/list.test.tsx [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+          ],
+        ]}
+        width="400px"
+      />
+    );
+  });
+  it('Renders list', () => {
+    expect(screen.getByTestId('list')).toBeDefined();
+  });
+  it('Renders items and titles', () => {
```
Comment (caarlosdamian) [inline src/components/list/list.test.tsx]: `  it('Render List component with custom props', () => `

File: src/components/list/list.test.tsx [hunk 19]
```diff
@@ -0,0 +1,30 @@
...
+          [
+            'This side of the list item should be around 50%',
+            'and this other side will be 50% fully',
+          ],
+        ]}
+        width="400px"
+      />
+    );
+  });
+  it('Renders list', () => {
```
Comment (caarlosdamian) [inline src/components/list/list.test.tsx]: `it('Render List component', () => {`

**Review by caarlosdamian — CHANGES_REQUESTED** [review]
Please check comments
# Please do not click resolve conversation just add 👍 to it then i will know is fix , i will resolve each comment

File: src/components/listItems/listItem.tsx (hunk 14 above)
Comment (thisisofficial) [reply src/components/listItems/listItem.tsx]: It means that it can either be a string or a number, if it aint any it will default to 'Nan', so the dev using this component that it is either empty or used incorrectly.

File: src/components/list/list.tsx (hunk 7 above)
Comment (thisisofficial) [reply src/components/list/list.tsx:11]: 👍

**Review by caarlosdamian — APPROVED** [review]
lgtm
